src/implementations/twig/components/button/button.story.js

```diff
@@ -11,117 +9,166 @@ import dataCall from '@ecl/specs-component-button/demo/data--call';
 import dataGhost from '@ecl/specs-component-button/demo/data--ghost';
 import dataSearch from '@ecl/specs-component-button/demo/data--search';
 
+import defaultSprite from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
 import button from './button.html.twig';
 import notes from './README.md';
 
 const iconsList = {};
-iconsList.none = '';
-
 uiIcons.forEach((icon) => {
   iconsList[icon] = icon;
 });
 
-// Preserve the adapted specs.
-const prepareButton = (data, type) => {
-  const typeLabel = type ? tabLabels.required : tabLabels.optional;
-  data.disabled = boolean('disabled', data.disabled, tabLabels.states);
-  data.label = text('label', data.label, tabLabels.required);
-
-  data.variant = select(
-    'variant',
-    [data.variant],
-    data.variant,
-    tabLabels.required
-  );
-
-  data.type = select('type', [data.type], data.type, typeLabel);
-
-  getExtraKnobs(data);
+const prepareControls = (data, args) => {
+  data.label = args.label;
+  data.disabled = args.disabled;
+  if (args.icon_name != null) {
+    data.icon = {};
+    data.icon.name = args.icon_name;
+    data.icon.type = 'ui';
+    data.icon.size = 'xs';
+    data.icon.transform = args.icon_transform;
+    data.icon.path = defaultSprite;
+    data.icon_position = args.icon_position;
+  } else if (args.icon_name == null && data.icon) {
+    delete data.icon;
+  }
 
   return data;
 };
 
 export default {
   title: 'Components/Button',
-  decorators: [withKnobs, withCode, withNotes],
+  argTypes: {
```

Comment:
`argTypes` seem to be the future, if the guys behind storybook place so much documentation and tools around it, then it will be the standard. We definitely need to make use of those.

It's pity that we don't have the react components anymore, there could have been inferred from the propTypes.

Reply:
I really like this structure way better than the knobs. When you think about it, it might not be impossible to extract schema from these getting the object keys + name + type + default value.